docs/developer-docs/security/formal-verification.md

```diff
@@ -0,0 +1,26 @@
+---
+sidebar_position: 4
+sidebar_label: Formal verification
+---
+# Formal verification
+
+Formal verification is, arguably, the highest form of quality assurance for software. Given a specification of what the system should do, formal verification tools check whether this specification is satisfied by a model of the system. The unique advantage of formal verification is that it can not only find bugs but also formally **prove** their absence – including the absence of security bugs. This goes beyond what testing or manual audits can achieve.
```

Comment:
```suggestion
Formal verification is the highest form of quality assurance for software. Given a specification of what the system should do, formal verification tools check whether this specification is satisfied by the system. The unique advantage of formal verification is that it can not only find bugs but also formally **prove** their absence – including the absence of security bugs. This goes beyond what testing or manual audits can achieve.
```

Reply:
It's important to make it clear that verification always checks a model. We do say it explicitly a couple of paragraphs later but I'd keep it to hammer it in. I'll remove the "arguably" part.